packages/fxa-auth-server/lib/payments/stripe.js

```diff
@@ -39,6 +38,36 @@ const stripe = require('stripe').Stripe;
  * @property {Plan['currency']} currency
  */
 
+/**
+ * Determine for two product metadata object's whether the new one
+ * is a valid upgrade for the old one.
+ *
+ * Throws errors if necessary metadata is not present to determine
+ * if its an upgrade.
+ *
+ * @param {AbbrevProduct['product_metadata']} oldMetadata Old product metadata
+ * @param {AbbrevProduct['product_metadata']} newMetadata New product metadata
+ * @returns {boolean} Whether the new product is an upgrade.
+ */
+function validateProductUpgrade(oldMetadata, newMetadata) {
+  if (!oldMetadata || !newMetadata) {
+    throw error.unknownSubscriptionPlan();
+  }
+
+  const oldId = oldMetadata.productSet;
+  const newId = newMetadata.productSet;
+  if (!oldId || oldId !== newId) {
+    // Incompatible product sets
+    return false;
+  }
+  const oldOrder = Number.parseInt(oldMetadata.productSetOrder);
+  const newOrder = Number.parseInt(newMetadata.productSetOrder);
```

Comment:
Since the default radix for `parseInt` is not 10, we should explicitly pass that.

Reply:
This is just code copied over from the subhub module that's already in production. Didn't want to mess with it too much